src/rust/ide/src/view/node_editor.rs

```diff
@@ -258,9 +258,25 @@ impl GraphEditorIntegratedWithControllerModel {
         Ok(())
     }
 
-    pub fn update_relevant_type_information
-    (&self, affected_asts:&[ExpressionId]) -> FallibleResult<()> {
-        debug!(self.logger, "Will update type information: {affected_asts:?}");
+    /// Set given type (or lack of such) on the given sub-expression.
+    fn set_type(&self, id:ExpressionId, typename:graph_editor::OptionalType) {
+        let event = (id,typename);
+        self.editor.frp.inputs.set_expression_type.emit_event(&event);
+    }
+
+    /// Look up the typename for the given expression in the execution controller's registry and
+    /// pass the data to the editor view.
+    fn update_type_on(&self, id:ExpressionId) {
+        let typename = self.lookup_typename(&id);
+        self.set_type(id,typename)
+    }
+
+    /// Like `update_type_on` but for multiple expressions.
+    fn update_types_on(&self, expressions_to_update:&[ExpressionId]) -> FallibleResult<()> {
+        debug!(self.logger, "Updating type information for IDs: {expressions_to_update:?}.");
+        for id in expressions_to_update {
+            self.update_type_on(*id)
+        }
         Ok(())
     }
```

Comment:
(Not crucial): In this section I started with the topmost function and then put functions used by it, etc. So please move those functions lower, or rearrange the whole section.